src/broadcastQueryClient-experimental/index.ts

```diff
@@ -0,0 +1,84 @@
+import { BroadcastChannel } from 'broadcast-channel'
+import { QueryClient } from '../core'
+
+interface BroadcastQueryClientOptions {
+  queryClient: QueryClient
+  broadcastChannel: string
+}
+
+export function broadcastQueryClient({
+  queryClient,
+  broadcastChannel = 'react-query',
+}: BroadcastQueryClientOptions) {
+  const channel = new BroadcastChannel(broadcastChannel, {
+    webWorkerSupport: false,
+  })
+
+  const queryCache = queryClient.getQueryCache()
+
+  queryClient.getQueryCache().subscribe((query, event) => {
+    if (!query) {
+      return
+    }
+
+    const { queryHash, queryKey, state } = query
+
+    if (
+      event?.eventType === 'dispatch' &&
+      (event?.dispatchAction?.type === 'success' ||
+        (event?.dispatchAction?.type === 'setState' &&
+          event?.dispatchAction?.setStateOptions?.meta?.source !== 'tabSync'))
+    ) {
+      channel.postMessage({
+        type: 'queryUpdated',
+        queryHash,
+        queryKey,
+        state,
+      })
+    }
+
+    if (event?.eventType === 'queryRemoved') {
+      channel.postMessage({
+        type: 'queryRemoved',
+        queryHash,
+        queryKey,
+      })
+    }
+  })
+
+  channel.onmessage = action => {
+    if (!action?.type) {
+      return
+    }
+
+    const { type, queryHash, queryKey, state } = action
+
+    if (type === 'queryUpdated') {
+      const query = queryCache.get(queryHash)
+
+      if (query) {
+        query.setState(state, {
```

Comment:
We might also be able to use a flag to indicate a transaction:
```ts
function broadcastQueryClient() {
  let transaction = false
  
  queryClient.getQueryCache().subscribe((query, event) => {
    if (transaction) {
      return
    }
  })
  
  channel.onmessage = action => {
    transaction = true
    query.setState(state)
    transaction = false
  }
}
```

Reply:
Great idea, done.